.github/workflows/pdm-lock-automation.yaml

```diff
@@ -66,30 +71,15 @@ jobs:
           fi
           source .venv/bin/activate
           pdm lock -G :all
-          git add pdm.lock
-
-      - name: Commit and push pdm.lock
-        if: steps.check.outputs.changed == 'true'
-        id: check_commit
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        run: |
-          if [ ! -d ".venv" ]; then
-            echo "Creating virtual environment inside root"
-            pdm venv create -w virtualenv --with-pip
-          else
-            echo "Virtual environment already exists."
-          fi
-
-          source .venv/bin/activate
-          pip install pre-commit~=3.6.2
-          pre-commit run pdm-lock-check
-
           if git diff --quiet; then
-            echo "No changes in compared to the SELF branch."
+            echo "No changes in compared to the ${{ github.head_ref }} branch."
             echo "Nothing to commit"
           else
-            echo "Changes detected in compared to the SELF branch."
+            git pull origin ${{ github.head_ref }} --allow-unrelated-histories --no-rebase
+            echo "Changes detected in compared to the ${{ github.head_ref }} branch."
```

Comment:
```suggestion
            echo "Changes detected between the current branch and the ${{ github.head_ref }} branch."
```

Reply:
@chandrasekharan-zipstack isn't the `github.head_ref` the current branch head?

Reply:
May be my understanding is wrong. Raising this to make sure.

Reply:
I would say print both `github.base_ref` and `github.head_ref`.

Check this doc for more details. https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/contexts#github-context

Reply:
@ritwik-g ${{ github.head_ref }} is correct.

github.head_ref: This variable refers to the name of the branch where the changes are coming from. In the context of a pull request, it’s the branch that contains the changes you want to merge into the base branch. For example, if you're creating a pull request from a branch named feature-branch into main, github.head_ref would be feature-branch.

github.base_ref: This variable refers to the name of the branch into which the changes are being merged. In the context of a pull request, it's the branch that will receive the changes. Continuing with the previous example, if you're merging feature-branch into main, github.base_ref would be main.

Reply:
@kirtimanmishrazipstack I agree with @ritwik-g on printing both the base and head ref rather than just going for `current` - it doesn't harm us in any way and it'll only make our lives slightly better while debugging anything on the worfklow